api/v1alpha7/openstackcluster_webhook.go

```diff
@@ -105,6 +105,11 @@ func (r *OpenStackCluster) ValidateUpdate(oldRaw runtime.Object) error {
 		r.Spec.ControlPlaneEndpoint = clusterv1.APIEndpoint{}
 	}
 
+	// Allow change only for the first time.
+	if old.Spec.DisableAPIServerFloatingIP && old.Spec.APIServerFixedIP == "" {
```

Comment:
Please can you add a unit test for this?

Reply:
Implemented with cb025df4.